src/java.base/unix/native/libjli/java_md_common.c

```diff
@@ -200,19 +200,23 @@ JLI_ReportErrorMessage(const char* fmt, ...) {
 JNIEXPORT void JNICALL
 JLI_ReportErrorMessageSys(const char* fmt, ...) {
     va_list vl;
-    char *emsg;
-
-    /*
-     * TODO: its safer to use strerror_r but is not available on
-     * Solaris 8. Until then....
-     */
-    emsg = strerror(errno);
-    if (emsg != NULL) {
-        fprintf(stderr, "%s\n", emsg);
-    }
 
     va_start(vl, fmt);
     vfprintf(stderr, fmt, vl);
+
+    if (errno != 0) {
+        /*
+         * Buffer size of 1024 copied from typical POSIX size used
+         * in strerror_r
+         */
+        char error[1024];
+        if(strerror_r(errno, error, sizeof error) == 0) {
```

Comment:
Style nit: need space after 'if'

This applies to other files too.

Reply:
Fixed, thanks for the heads up